src/enrichment/lsst.rs

```diff
@@ -68,12 +69,99 @@ pub fn create_lsst_alert_pipeline() -> Vec<Document> {
+        // TODO: Do this properly
+        let kafka_producer_host = config
+            .get_string("kafka.producer")
+            .unwrap_or_else(|_| "broker:29092".to_string());
+
+        // Create Kafka producer
+        let kafka_producer: rdkafka::producer::FutureProducer =
+            rdkafka::config::ClientConfig::new()
+                // Uncomment the following to get logs from kafka (RUST_LOG doesn't work):
+                // .set("debug", "broker,topic,msg")
+                .set("bootstrap.servers", &kafka_producer_host)
+                .set("message.timeout.ms", "5000")
+                // it's best to increase batch.size if the cluster
+                // is running on another machine. Locally, lower means less
+                // latency, since we are not limited by network speed anyways
+                .set("batch.size", "16384")
+                .set("linger.ms", "5")
+                .set("acks", "1")
+                .set("max.in.flight.requests.per.connection", "5")
+                .set("retries", "3")
+                .create()
+                .unwrap();
+        Babamul { kafka_producer }
+    }
+
+    async fn process_alert(
+        &self,
+        alert: Document,
+        properties: Document,
+    ) -> Result<(), EnrichmentWorkerError> {
+        // Merge properties into the alert document
+        let mut merged = alert.clone();
+        merged.insert("properties", Bson::Document(properties.clone()));
+
+        // Determine if this alert is worth sending to Babamul
+        let min_reliability = 0.5;
+        let pixel_flags = vec![1, 2, 3];
+        let sso = false;
+
+        if (merged.get_f64("candidate.reliability").unwrap_or(0.0) < min_reliability)
+            || (merged
+                .get_document("candidate")
+                .unwrap()
+                .get_i32("pixel_flags")
+                .map(|pf| pixel_flags.contains(&pf))
+                .unwrap_or(false))
+            || (merged
+                .get_document("properties")
+                .unwrap()
+                .get_bool("rock")
+                .unwrap_or(false)
+                == sso)
+        {
+            // Skip this alert, it doesn't meet the criteria
+            return Ok(());
+        }
+
+        // Determine which topic this alert should go to
+        // Is it a star, galaxy, or none, and does it have a ZTF crossmatch?
+        // TODO: Get this implemented
+        // For now, all LSST alerts go to "babamul.none"
+        let category: String = "none".to_string();
+        let topic_name = format!("babamul.{}", category);
+        let schema = crate::filter::load_alert_schema().unwrap();
+        // Convert the merged document into an Alert
+        let alert_obj = Alert::from_bson_document(&merged).unwrap();
+        let _ = crate::filter::send_alert_to_kafka(
+            &alert_obj,
+            &schema,
+            &self.kafka_producer,
+            &topic_name,
+        )
+        .await
+        .unwrap();
```

Comment:
We definitely want to send batches of alerts to kafka rather than one a time, that will add a ton of latency to this worker. I believe we do the batch stuff in the filter worker already? Best would be to do the same thing, so your babamul worker here could do the vetting of the alerts and formatting and return an Option<whateverisyouroutput> that you add to a vec in the main loop of the enrichment worker, and push to kafka as a batch if the vec isn't empty. Since we'll have multiple topics perhaps best is to use a hashmap rather than a vec.

Reply:
Just implemented this. We'll still need to decide on the Avro schema, but there's a placeholder for now.